.github/workflows/ci.yaml

```diff
@@ -100,6 +104,12 @@ jobs:
         name: unit-test-results
         path: LoRaEngine/test/TestResults
 
+    - name: Upload to Codecov
+      run: |
+        curl -OsSL https://uploader.codecov.io/v0.1.0_4653/linux/codecov
+        chmod +x codecov
+        ./codecov
+
```

Comment:
Seems a bit weird to have to do this with CURL-ing a specific version especially and making it executable because of updates - we would have to bump up the version manually here. We could use [the latest always](https://uploader.codecov.io/linux/latest) to avoid this but shouldn't we use a [Github action for this](https://github.com/marketplace/actions/codecov#usage) - v2 is not deprecated if I understand correctly.

Reply:
Also, the token is not provided anywhere, I assume because Azure (the org) has given permission to the Codecov application?

Reply:
Considering that it's version 0._x_, I assume that there can be breaking changes between minor versions and therefore would prefer to stick with a version known to be working and upgrade manually when needed. Perhaps one can review this when it goes 1._x_. 

Downloading the binary, making it executable and then executing it seemed simple enough that it didn't seem to warrant the use of a whole GitHub Action when the usage was already so straightforward (and frankly what I was used to doing from other projects when the action did not exist). That said, [I've made a note](https://github.com/Azure/iotedge-lorawan-starterkit/projects/8#card-69845696) on the [backlog](https://github.com/Azure/iotedge-lorawan-starterkit/projects/8) to consider using the GitHub Action.

Reply:
I don't see it the same way: since we have the tools (Github action) we should use them. We can resolve here since this is a small/insignificant example but if we apply the same principle in general we will re-invent a lot of things, just food for thought.

Reply:
Same here. I think since there is an action, it seems easier to use that to me.

Reply:
I had already moved to the action version in PR #432, shortly after wrapping up this one. It is indeed simpler.